### ISSUE TITLE
Crash when updating underlying workspace for a MantidMatrix

### ISSUE THREAD

**Issue by gemmaguest** [mantidproject/mantid#21085]
### Expected behavior
A MantidMatrix should be updated to reflect its underlying workspace and should not crash.

### Actual behavior
- A MantidMatrix for a workspace that does NOT contain X errors is not updated to include an X errors tab if the underlying workspace is changed to contain X errors.
- Mantid can crash if a MantidMatrix is open and displaying an X errors tab when the underlying workspace is updated to remove the X errors.

### Steps to reproduce the behavior
- Run this script to create a workspace without X errors:
```
CreateWorkspace(OutputWorkspace='ws', DataX='1,2', DataY='10.123', DataE='0.5', NSpec=1, UnitX='TOF')
```
- Double click the `ws` to show the MantidMatrix and expand it so you can see all the tabs. It does not include an X errors tab.
- Run this script to replace the workspace with one that contains X errors. Note that the Mantid Matrix does not update to show the X errors tab.
```
CreateWorkspace(OutputWorkspace='ws', DataX='1,2', DataY='10.123', DataE='0.5', NSpec=1, UnitX='TOF')
for i in range(0,len(mtd['ws'].dataDx(0))):
    mtd['ws'].dataDx(0)[i] = 0.5 
```
- Close and re-open `ws`. The X errors tab should now appear. Select it.
- Run this script to replace the workspace with one that does not contain X errors. Mantid crashes.
```
CreateWorkspace(OutputWorkspace='ws', DataX='1,2', DataY='10.123', DataE='0.5', NSpec=1, UnitX='TOF')
```

### Platforms affected
Ubuntu 16.04, possibly all

**Comment by gemmaguest** [comment]
This might be worth looking into in conjunction with #19096, which is also to do with updating MantidMatrix

**Comment by gemmaguest** [comment]
The crash seems to be fixed in v6.0.0. The tabs are still not updated but closing this as stale as we're not likely to get around to fixing it soon
